### PR TITLE
feat(tracing): Allow transaction renaming [NATIVE-438]

### PR DESCRIPTION
**Pull request by loewenheim**

### CODE REVIEW

**Review by Swatinem — APPROVED** [review]

File: src/sentry_value.c [hunk 1]
```diff
@@ -1326,3 +1326,9 @@ sentry_transaction_remove_data(sentry_value_t transaction, const char *key)
 {
     sentry_span_remove_data(transaction, key);
 }
+
+void
+sentry_transaction_set_name(sentry_value_t tx, const char *name)
+{
+    sentry_value_set_by_key(tx, "transaction", sentry_value_new_string(name));
```
Comment (flub) [inline src/sentry_value.c:1333]: this can return an error if tx was not an object...  is it really a good idea to make this void?

Comment (loewenheim) [reply src/sentry_value.c:1333]: This problem should solve itself once we have strong types.

Comment (flub) [reply src/sentry_value.c:1333]: true, fain enough

Comment (flub) [reply src/sentry_value.c:1333]: *fair

**Pull request merged**